soepy/python/pre_processing/model_processing.py

```diff
@@ -38,7 +38,28 @@ def expand_init_dict(init_dict):
     ]
     shocks_cov = [shocks_cov[0] ** 2, shocks_cov[1] ** 2, shocks_cov[2] ** 2]
 
-    init_dict["DERIVED_ATTR"] = {"educ_range": educ_range, "shocks_cov": shocks_cov}
+    # Extract the number of types
+
+    if [v for k, v in init_dict["PARAMETERS"].items() if "share" in k] == 1.0:
```

Comment:
The expression `v for k, v in init_dict["PARAMETERS"].items() if "share" in k` is constructed numerous times here, please create one auxiliary variable instead. Also, please add some comment here what is going on. It appears too complicated for what we want to achieve, at least at first glance :)

Reply:
Creating a variable is definitely a very good idea for increasing readability. Done. In case you see a way to do the whole thing in an uneasier way in the more readable version, please share. I do tend to think in complicated ways sometimes.